pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2430,11 +2429,41 @@ public void resetSegment(String tableNameWithType, String segmentName, @Nullable
     }
   }
 
+  /**
+   * Resets segments with Error EV of a table. This operation invoke resetPartition via state transition message.
+   */
+  public void resetErrorEvSegments(String tableNameWithType, @Nullable String targetInstance) {
```

Comment:
Same here, we may introduce a boolean `errorSegmentsOnly` to `resetAllSegments`

Reply:
make sense, updated